src/pages/index.js

```diff
@@ -0,0 +1,380 @@
+  console.log(cardId);
+  openModal(deleteModal);
+}
+
+function handleLike(evt, id) {
+  // 1. Check if the card is currently liked or not
+  const isLiked = evt.target.classList.contains("card__like-button_liked");
+  console.log(id);
+  // 2. Call the changeLikeStatus method passing the appropriate arguments
+  api
+    .changeLikeStatus(id, !isLiked) // Pass the opposite of the current state
+    .then((data) => {
+      console.log(data);
+      // 3. Handle the response: toggle the 'liked' class based on the updated status
+      evt.target.classList.toggle("card__like-button_liked", data.isLiked);
+    })
+    .catch(console.error); // 4. Handle any errors in the catch block
+}
+
+function getCardElement(data) {
+  console.log(cardTemplate);
+  const cardElement = cardTemplate.content
+    .querySelector(".card")
+    .cloneNode(true);
+
+  const cardNameElement = cardElement.querySelector(".card__title");
+  const cardImageElement = cardElement.querySelector(".card__image");
+  const cardLikeButton = cardElement.querySelector(".card__like-button");
+  const deleteButton = cardElement.querySelector(".card__delete-button");
+
+  cardNameElement.textContent = data.name;
+  cardImageElement.src = data.link;
+  cardImageElement.alt = data.name;
+
+  if (data.isLiked) {
+    cardLikeButton.classList.add("card__like-button_liked");
+  }
+
+  cardLikeButton.addEventListener("click", () => {
+    console.log("Card ID before liking:", data._id);
+    cardLikeButton.classList.toggle("card__like-button_liked");
+  });
+
+  cardImageElement.addEventListener("click", () => {
+    previewModalImageElement.src = data.link;
+    previewModalCaptionElement.textContent = data.name;
+    previewModalImageElement.alt = data.name;
+    openModal(previewModal);
+  });
+  cardLikeButton.addEventListener("click", (evt) => handleLike(evt, data._id));
+
+  deleteButton.addEventListener("click", (evt) => {
+    handleDeleteCard(evt, cardElement, data._id);
+  });
+
+  modalCancelButton.addEventListener("click", () => {
+    console.log("delete cancelled");
+    closeModal(deleteModal);
+  });
+
+  return cardElement;
+}
+
+function handleOverlayClick(evt) {
+  // Ensure the click is on the overlay, not the modal content
+  if (evt.target === evt.currentTarget) {
+    closeModal(evt.currentTarget);
+  }
+}
+
+function handleEscape(evt) {
+  console.log("called");
+  console.log(evt.key);
+  if (evt.key === "Escape") {
+    const activeModal = document.querySelector(".modal_opened");
+    console.log(activeModal);
+    closeModal(activeModal);
+  }
+}
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+
+  document.addEventListener("keyup", handleEscape);
+  modal.addEventListener("click", handleOverlayClick);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keyup", handleEscape);
+  modal.removeEventListener("click", handleOverlayClick);
+}
+
+function disableButton(button, settings) {
```

Comment:
Please, delete `disableButton` from `index.js` because you have `disableButton` inside `validation.js`. You should import it from there and use it in `index.js`